### ISSUE TITLE
Installing problem

### ISSUE THREAD

**Issue by gxgaga** [BinaryAnalysisPlatform/bap#320]
Hi, 

I got some problems installing the BAP.

When I tried to install on Mac OS X, it reports, 

![image](https://cloud.githubusercontent.com/assets/4698662/12097700/15e83420-b2eb-11e5-878b-72418e83d56c.png)

When I tried to install on Ubuntu, it returns,

![image](https://cloud.githubusercontent.com/assets/4698662/12097691/0a0c5a14-b2eb-11e5-8a2c-9ef0d387cc95.png)

No idea. So.... Thanks!

Xing

**Comment by ivg** [comment]
On Ubuntu you have to old, incompatible C++ compiler. Ideally, BAP should be compiled with clang 3.4 or later. Also, you're installing a very old `bap` (that's unrelated, but still suspicious), so don't forget to hit `opam update` before installing it. 

Ideally, when you say `opam depext bap` it should install correct external dependencies and compilers. If it didn't then it is a separate issue.

In short, install `clang++` on both systems. In Mac OS X, you should have a complete development environment set up. We're using macports, so I would recommend to install it following the standard instructions. Then you can use `opam depext` that will take care of system packages.

**Comment by ivg** [comment]
Btw, what version of mac os x and ubuntu are you using?

**Comment by gxgaga** [comment]
Thanks for the reply. Ubuntu is a little bit old, it is 12.04. But OS X is
Yosemite. Also I did use macports instead of homebrew. I may try it
again...

2016-01-05 14:24 GMT-05:00 Ivan Gotovchits <notifications@github.com>:

**Comment by ivg** [comment]
feel free to reopen, if you still have any issues
